### PR TITLE
Jane/PTKB-4682/grafana tls

### PR DESCRIPTION
**Pull request by jane-prowler**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md

### CODE REVIEW

File: stable/grafana/templates/NOTES.txt [hunk 1]
```diff
@@ -20,9 +20,9 @@
    NOTE: It may take a few minutes for the LoadBalancer IP to be available.
         You can watch the status of by running 'kubectl get svc --namespace {{ .Release.Namespace }} -w {{ template "grafana.fullname" . }}'
      export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "grafana.fullname" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
-     http://$SERVICE_IP:{{ .Values.service.port -}}
```
Comment (aggronerd) [inline stable/grafana/templates/NOTES.txt:23]: This may be a dumb question: where does the grafana.ini file come from?

Comment (aggronerd) [reply stable/grafana/templates/NOTES.txt:23]: ...and how does it differ from .Values?

File: stable/grafana/templates/NOTES.txt [hunk 2]
```diff
@@ -20,9 +20,9 @@
...
+     {{ (index .Values "grafana.ini").server.protocol }}://$SERVICE_IP:{{ .Values.service.port -}}
 {{ else if contains "ClusterIP"  .Values.service.type }}
-     export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "grafana.fullname" . }},component={{ .Values.name }}" -o jsonpath="{.items[0].metadata.name}")
+     export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "grafana.name" . }}" -o jsonpath="{.items[0].metadata.name}")
```
Comment (aggronerd) [inline stable/grafana/templates/NOTES.txt:25]: Another dumb question perhaps, but why was this changed?

File: stable/grafana/templates/deployment.yaml [hunk 3]
```diff
@@ -106,6 +106,10 @@ spec:
             - name: ldap
               mountPath: "/etc/grafana/ldap.toml"
               subPath: ldap.toml
+          {{- if .Values.tls.enabled }}
+            - name: tls
+              mountPath: {{ .Values.tls.certDirectory }}
```
Comment (aggronerd) [inline stable/grafana/templates/deployment.yaml:111]: Maybe specify a default for this path regardless?

**Review by aggronerd — COMMENTED** [review]
Looks good to me. Is there a readme we need to update to explain how to optionally use this?

File: stable/grafana/templates/NOTES.txt (hunk 2 above)
Comment (jane-prowler) [reply stable/grafana/templates/NOTES.txt:25]: This is a bug in the official chart.  The pods aren't labelled with component, and the app they are labelled with is from grafana.name not grafana.fullname

File: stable/grafana/templates/NOTES.txt (hunk 1 above)
Comment (jane-prowler) [reply stable/grafana/templates/NOTES.txt:23]: Let's discuss offline.

File: stable/grafana/templates/deployment.yaml (hunk 3 above)
Comment (jane-prowler) [reply stable/grafana/templates/deployment.yaml:111]: Let's discuss offline

**Review by aggronerd — APPROVED** [review]
